### PR TITLE
Visual Resources for Storage Objects

### LINKED ISSUE #1048: Visual Resources for Storage Objects
As [Sebastian](https://github.com/kyma-project/community/blob/main/manifesto/personas.md#sebastian---tech-lead) and [Mathew](https://github.com/kyma-project/community/blob/main/manifesto/personas.md#matthew---cloud-native-software-engineer) I would like to see in a visual way the connections for our Storage Objects.

**Acceptance Criteria**

- [ ] Follow Project DOD
- [x] We will follow Fiori Guidelines when comes to the style of the resource boxes
- [x] We will follow just 1 level of dependency
- [x] We Cover Secrets, ConfigMaps,PVC, PV 
- [x] The Styling is done in a separate task

**Tasks**

- [ ] 

**Inspiration**

- [Storage Objects](https://shipit.dev/posts/kubernetes-overview-diagrams.html#storage)

image

### PR DESCRIPTION
**Pull request by mrCherry97**
**Description**

Changes proposed in this pull request:

- ...
- ...
- ...

**Related issue(s)**
Closes #1048

### CODE REVIEW

File: core-ui/src/components/Predefined/Create/OAuthClients/OAuthClients.create.js [hunk 1]
```diff
@@ -188,5 +188,9 @@ OAuth2ClientsCreate.resourceGraphConfig = (t, context) => ({
       kind: 'Secret',
     },
   ],
+  matchers: {
+    OAuth2Client: (client, secret) =>
```
Comment (Wawrzyn321) [inline core-ui/src/components/Predefined/Create/OAuthClients/OAuthClients.create.js:192]:
```suggestion
    Secret: (client, secret) =>
```
We would match OAuth2Client with itself.

File: core-ui/src/components/Predefined/Create/ServiceAccounts/ServiceAccounts.create.js [hunk 2]
```diff
@@ -155,5 +155,10 @@ ServiceAccountsCreate.resourceGraphConfig = (t, context) => ({
   ],
   depth: 2,
   networkFlowLevel: 2,
+  matchers: {
+    Secret: (sa, secret) =>
+      sa.secrets?.find(s => s.name === secret.metadata.name) ||
+      sa.imagePullSecrets?.find(s => s.name === secret.metadata.name),
+  },
```
[the remark this answers is not in the document]
Comment (mrCherry97) [reply core-ui/src/components/Predefined/Create/ServiceAccounts/ServiceAccounts.create.js:162]: We have to discuss the convention for this. I think now the code is more clear and "easier to read", but I'm open to talk about it.

File: core-ui/src/components/Predefined/Create/PersistentVolumeClaims/PersistentVolumeClaims.create.js [hunk 3]
```diff
@@ -30,4 +30,20 @@ function PersistentVolumeClaimsCreate({
...
+    },
+    {
+      kind: 'StorageClass',
+    },
+  ],
+  depth: 1,
+  networkFlowLevel: 1,
+  matchers: {
+    StorageClass: (pvc, sc) => pvc.spec.storageClassName === sc.metadata.name,
+  },
```
Comment (Wawrzyn321) [inline core-ui/src/components/Predefined/Create/PersistentVolumeClaims/PersistentVolumeClaims.create.js:46]: It looks like we could connect also `PersistentVolumes` here.

**Review by Wawrzyn321 — APPROVED** [review]

**Pull request merged**